lib/post_destroyer.rb

```diff
@@ -196,6 +198,16 @@ def trash_public_post_actions
     Post.with_deleted.where(id: @post.id).update_all(Hash[*f.flatten])
   end
 
+  def recover_public_post_actions(post_deleted_at)
+    PostAction.publics
+      .where(post_id: @post.id)
+      .with_deleted
+      .where(deleted_at: post_deleted_at..(post_deleted_at + 5.seconds))
+      .update_all(deleted_at: nil, deleted_by_id: nil)
+
+    @post.post_actions.each(&:update_counters)
```

Comment:
We need to test that the counters are updated.

Reply:
@ZogStriP Hmm if we need to update the counters here anyway, it'll probably be better to just do `find_each` and call `recover!`

Reply:
@pmusaraj can you find a way to update the counters once per `post_action_type`?

Reply:
@ZogStriP I couldn't find a way to update the counters once for `post_action_type`. I tried this below: 

    PostActionType.public_types.each do |z, type_id|
      @post.post_actions.where(post_action_type_id: type_id).last(&:update_counters)
    end

but it didn't work. So I went back to `find_each` and `recover!` as Alan suggested.